### PR TITLE
add ecila keyboard

### PR DESCRIPTION
**Pull request by mincedshon**
added ecila files

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by keyboard-magpie** [comment]
If you are anticipating revisions the more usual file structure would be to have common configuration in the upper level keyboard folder then the rev1 specific files in a rev1 folder, example here https://github.com/qmk/qmk_firmware/tree/master/keyboards/keebio/iris

**Comment by drashna** [comment]
Please do not close and reopen new PRs: 
* #17406 
* #17580

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
> If you are anticipating revisions the more usual file structure would be to have common configuration in the upper level keyboard folder then the rev1 specific files in a rev1 folder, 

To expand on this, if you don't plan on having revisions, you should remove the folder and it's settings into the folder above.  

Also, if you do plan on releasing additional boards/pcbs, it would be a good idea to use a vendor folder now.   Eg, `keyboards/mincedshon/ecila`

**Comment by mincedshon** [comment]
> Please do not close and reopen new PRs:
> 
> * [ecila keyboard #17406](https://github.com/qmk/qmk_firmware/pull/17406)
> * [add ecila keyboard #17580](https://github.com/qmk/qmk_firmware/pull/17580)
> 
> If you're having troubles or need help with git stuff, please ask! We're here to help, especially as git can be ... tricky.

sorry about that, i was working on the master branch for the previous 2 PRs

**Review by keyboard-magpie — APPROVED** [review]

File: keyboards/mincedshon/ecila/info.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
+{
+    "keyboard_name": "ecila",
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/info.json:2]:
```suggestion
    "keyboard_name": "ecila",
    "manufacturer": "mincedshon",
```

File: keyboards/mincedshon/ecila/info.json [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    "maintainer": "mincedshon",
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/info.json:3]:
```suggestion
    "maintainer": "mincedshon",
    "usb": {
        "vid": "0x534B",
        "pid": "0x534B",
        "device_version": "0.0.1",
    },
```

File: keyboards/mincedshon/ecila/readme.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# ecila
+
+*An Alice layout with arrow cluster*
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/readme.md:3]:
```suggestion
An Alice layout with arrow cluster
```

File: keyboards/mincedshon/ecila/config.h [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x534B
+#define PRODUCT_ID      0x534B
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    mincedshon
+#define PRODUCT         ecila
+
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/config.h:14]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mincedshon/ecila/config.h [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x534B
+#define PRODUCT_ID      0x534B
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    mincedshon
+#define PRODUCT         ecila
```
Comment (keyboard-magpie) [inline keyboards/mincedshon/ecila/config.h:13]: As you have missed PR  #17815 going through, can you please move these parameters to your info.json? Please reference the PR for how this is done.

**Review by keyboard-magpie — CHANGES_REQUESTED** [review]

File: keyboards/mincedshon/ecila/config.h (hunk 4 above)
Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h]: thank you fauxpark, the changes have been made!

File: keyboards/mincedshon/ecila/config.h (hunk 5 above)
Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h]: the changes have been made, thank you!

File: keyboards/mincedshon/ecila/info.json [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{
+    "keyboard_name": "ecila",
+    "manufacturer": "mincedshon",
+    "maintainer": "mincedshon",
+    "usb": {
+      "vid": "0x534B",
+      "pid": "0x534B",
+      "device_version": "0.0.1",
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/info.json:8]:
```suggestion
      "device_version": "0.0.1"
```
Oops

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/mincedshon/ecila/config.h [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+/* key matrix pins */
+#define MATRIX_ROW_PINS { E6, B3, B5, B4, D7 }
+#define MATRIX_COL_PINS { B6, F7, F6, F5, F4, F1, F0, D6, B7, D0, D1, D2, D3, D5, D4 }
+#define UNUSED_PINS
+
+/* COL2ROW or ROW2COL */
+#define DIODE_DIRECTION COL2ROW
+
+/* Set 0 if debouncing isn't needed */
+#define DEBOUNCE 5
```
Comment (fauxpark) [inline keyboards/mincedshon/ecila/config.h:31]: IF you want, you can specify these in info.json as well.

**Review by fauxpark — APPROVED** [review]

File: keyboards/mincedshon/ecila/config.h (hunk 7 above)
Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h:31]: may i know the reason behind this?? am a little lost on why the IDs need to be moved too

Comment (drashna) [reply keyboards/mincedshon/ecila/config.h:31]: because we're moving to a data driven model, where most of the info is in the info.json.

Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h:31]: ah okay, will do so!

Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h:31]: do lines 18-31 translate to this in info.json?

    "MATRIX_ROWS": 5,
    "MATRIX_COLS": 15,
    "MATRIX_ROW_PINS": { E6, B3, B5, B4, D7 },
    "MATRIX_COL_PINS": { B6, F7, F6, F5, F4, F1, F0, D6, B7, D0, D1, D2, D3, D5, D4 },
    "UNUSED_PINS": ,
    "DIODE_DIRECTION": COL2ROW,
    "DEBOUNCE": 5,

Comment (fauxpark) [reply keyboards/mincedshon/ecila/config.h:31]: https://docs.qmk.fm/#/reference_info_json

Comment (mincedshon) [reply keyboards/mincedshon/ecila/config.h:31]: thank you!

**Comment by mincedshon** [comment]
able to merge?

**Review by fauxpark — APPROVED** [review]

**Pull request merged**